src/lazy/encoder/binary/value_writer.rs

```diff
@@ -0,0 +1,886 @@
+        if encoded_length <= 13 {
+            type_descriptor |= encoded_length as u8;
+            self.push_byte(type_descriptor);
+        } else {
+            type_descriptor |= 0xEu8;
+            self.push_byte(type_descriptor);
+            VarUInt::write_u64(self.encoding_buffer, encoded_length as u64)?;
+        }
+
+        self.push_bytes(magnitude_be_bytes.as_slice());
+
+        Ok(())
+    }
+
+    pub fn write_f32(mut self, value: f32) -> IonResult<()> {
+        if value == 0f32 && !value.is_sign_negative() {
+            self.push_byte(0x40);
+            return Ok(());
+        }
+
+        self.push_byte(0x44);
+        self.push_bytes(&value.to_be_bytes());
+        Ok(())
+    }
+
+    pub fn write_f64(mut self, value: f64) -> IonResult<()> {
+        if value == 0f64 && !value.is_sign_negative() {
+            self.push_byte(0x40);
+            return Ok(());
+        }
+
+        self.push_byte(0x48);
+        self.push_bytes(&value.to_be_bytes());
+        Ok(())
+    }
+
+    pub fn write_decimal(self, value: &Decimal) -> IonResult<()> {
+        let _encoded_size = self.encoding_buffer.encode_decimal_value(value)?;
+        Ok(())
+    }
+
+    pub fn write_timestamp(self, value: &Timestamp) -> IonResult<()> {
+        let _ = self.encoding_buffer.encode_timestamp_value(value)?;
+        Ok(())
+    }
+
+    pub fn write_string<A: AsRef<str>>(mut self, value: A) -> IonResult<()> {
+        let text: &str = value.as_ref();
+        let encoded_length = text.len(); // The number of utf8 bytes
+
+        let type_descriptor: u8;
+        if encoded_length <= MAX_INLINE_LENGTH {
+            type_descriptor = 0x80 | encoded_length as u8;
+            self.push_byte(type_descriptor);
+        } else {
+            type_descriptor = 0x8E;
+            self.push_byte(type_descriptor);
+            VarUInt::write_u64(self.encoding_buffer, encoded_length as u64)?;
+        }
+        self.push_bytes(text.as_bytes());
+        Ok(())
+    }
+
+    pub fn write_symbol<A: AsRawSymbolTokenRef>(self, value: A) -> IonResult<()> {
+        match value.as_raw_symbol_token_ref() {
+            RawSymbolTokenRef::SymbolId(sid) => self.write_symbol_id(sid),
+            RawSymbolTokenRef::Text(text) => IonResult::illegal_operation(format!(
+                "the Ion 1.0 raw binary writer cannot write text symbols (here: '{text}')"
+            )),
+        }
+    }
+
+    pub fn write_clob<A: AsRef<[u8]>>(self, value: A) -> IonResult<()> {
+        let bytes: &[u8] = value.as_ref();
+        // The clob type descriptor's high nibble is type code 9
+        self.write_lob(bytes, 0x90)
+    }
+
+    pub fn write_blob<A: AsRef<[u8]>>(self, value: A) -> IonResult<()> {
+        let bytes: &[u8] = value.as_ref();
+        // The blob type descriptor's high nibble is type code 10 (0xA)
+        self.write_lob(bytes, 0xA0)
+    }
+
+    fn list_writer(&mut self) -> BinaryListWriter_1_0<'_, 'top> {
```

Comment:
Yes it should, thanks.

Reply:
On closer inspection, this method is right and the other two are wrong. They no longer return the `IonResult<_>` wrapper because they can't fail and they're not trying to conform to a trait signature.